### PR TITLE
feat(samples): parameterize storageclass for the e2e-mnist example

### LINKED ISSUE #270: Update e2e-mnist example with configurable storageclass
/kind bug

**What steps did you take and what happened:**
The [e2e-mnist](https://github.com/kubeflow/kfp-tekton/tree/master/samples/e2e-mnist) example was originally created to use the default storageclass so it won't tie to any cloud provider. However, since many IBM Cloud users are not familiar with ibm cloud file system and many of the original KFP examples are required with GCP, we should expose more IBM Cloud concepts in our example since majority of the kfp-tekton users are using IBM Cloud.

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #270 

**Description of your changes:**
Add storageclass as one of the configurable parameter for e2e-mnist example. This way is more flexible to change storageclass on a per pipeline basis.

**Environment tested:**

* Python Version (use `python --version`): 3.7
* Tekton Version (use `tkn version`): 1.15
* Kubernetes Version (use `kubectl version`): 1.16
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

File: samples/e2e-mnist/mnist.ipynb [hunk 1]
```diff
@@ -13,9 +13,11 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Define the model name and Kubeflow user namespace\n",
+    "### Define the model name, Kubeflow user namespace, and storageclass for running this pipeline\n",
     "\n",
-    "Change the below cell with a name you want to use for this pipeline and which namespace you want to execute on Kubeflow."
+    "Change the below cell with a name you want to use for this pipeline, which namespace you want to execute on Kubeflow, and the [storageclass](https://kubernetes.io/docs/concepts/storage/storage-classes/) to run and store the trained model. **Make sure the storageclass below supports ReadWriteMany in order to run distributed training.** \n",
```
Comment (shawnzhu) [inline samples/e2e-mnist/mnist.ipynb:18]: Do you mind adding this as one prerequisite in `README.md`?

Comment (Tomcli) [reply samples/e2e-mnist/mnist.ipynb:18]: done

File: samples/e2e-mnist/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ To run this pipeline, make sure your cluster has at least 16 cpu and 32GB in tot
     ```shell
     pip install jupyter numpy Pillow
     ```
-
+- Make sure the Kubernetes Cluster has a [storageclass](https://kubernetes.io/docs/concepts/storage/storage-classes/) that supports ReadWriteMany in order to run distributed training.
```
Comment (shawnzhu) [inline samples/e2e-mnist/README.md:15]: 👍

File: samples/e2e-mnist/mnist.ipynb [hunk 3]
```diff
@@ -13,9 +13,11 @@
...
+    "\n",
+    "For any cloud provider such as AWS and GCP, you can find out the default storageclass by running the bash command `kubectl get storageclass` and replace the storageclass variable below. If the Kubernetes Cluster doesn't have any storageclass, simply assign the storageclass variable below as `''` to use the generic Kubernetes persistent storage claim."
```
Comment (animeshsingh) [inline samples/e2e-mnist/mnist.ipynb:20]:
```suggestion
    "The default storageclass value below will work on IBM Cloud. For other cloud providers such as AWS and GCP, find out the default storageclass by running the bash command `kubectl get storageclass`, and replace the storageclass variable default value below. If the Kubernetes Cluster doesn't have any storageclass, simply assign the storageclass variable below as `''` to use the generic Kubernetes persistent storage claim."
```

**Pull request merged**